src/complex/DataStructure/DataArray.hpp

```diff
@@ -496,6 +500,61 @@ class DataArray : public IDataArray
     return *this;
   }
 
+  /**
+   * @brief getTypeName
+   * @return
+   */
+  std::string getTypeName() const override
+  {
+    std::string typeName = "UNKNOWN";
+
+    if constexpr(std::is_same_v<T, int8_t>)
+    {
+      typeName = "DataArray<int8_t>";
+    }
+    else if constexpr(std::is_same_v<T, uint8_t>)
+    {
+      typeName = "DataArray<uint8_t>";
+    }
+    else if constexpr(std::is_same_v<T, int16_t>)
+    {
+      typeName = "DataArray<int16_t>";
+    }
+    else if constexpr(std::is_same_v<T, uint16_t>)
+    {
+      typeName = "DataArray<uint16_t>";
+    }
+    else if constexpr(std::is_same_v<T, int32_t>)
+    {
+      typeName = "DataArray<int32_t>";
+    }
+    else if constexpr(std::is_same_v<T, uint32_t>)
+    {
+      typeName = "DataArray<uint32_t>";
+    }
+    else if constexpr(std::is_same_v<T, int64_t>)
+    {
+      typeName = "DataArray<int64_t>";
+    }
+    else if constexpr(std::is_same_v<T, uint64_t>)
+    {
+      typeName = "DataArray<uint64_t>";
+    }
+    else if constexpr(std::is_same_v<T, float>)
+    {
+      typeName = "DataArray<float>";
+    }
+    else if constexpr(std::is_same_v<T, double>)
+    {
+      typeName = "DataArray<double>";
+    }
+    else if constexpr(std::is_same_v<T, bool>)
+    {
+      typeName = "DataArray<bool>";
+    }
+    return typeName;
```

Comment:
Why is this needed? int8 and int8_t should resolve to the same thing? This will also have ramifications on reading/writing the dream3d files as the typeName is written as a piece of meta-data that is then used during reading to determine what kind of DataArray to create.